### DISCUSSION TITLE
Nix Package

### DISCUSSION THREAD

**Discussion by Modularius** [taosdata/TDengine#23296] [Q&A]
Has anyone written a Nix package for tdengine? Or has anyone looked into this? I don't have much experience with Nix, but I've tried myself, however I run into problems building some of the external libraries.

Sorry for the vague description, I don't want to go into great detail if someone has already successfully done this.

Thanks

**Comment by aloscha1** [comment]
There probably were people who tried it but I'm not aware of any existing Nix packages for TDengine. Nix Packaging for a complex software like TDengine is quite challenging especially if there are dependencies or build issues with external libraries.
I think you should search forums and look at repositories related to Nix packages and see if anyone had a similar problem to yours.

Good luck!
